### PR TITLE
[ruby] Improve ruby client examples

### PR DESCRIPTION
**Pull request by zippolyte**
Improve ruby client examples for models, by building models with required properties set.
Goal is to have examples a bit more copy/pastable directly
Adds `constructExampleCode` methods, similarly to a few other clients like go, or powershell

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [ ] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RubyClientCodegen.java [hunk 1]
```diff
@@ -667,4 +569,178 @@ protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel, Sc
...
+
+    private String constructExampleCode(CodegenProperty codegenProperty, HashMap<String, CodegenModel> modelMaps, HashMap<String, Integer> processedModelMap) {
+        if (codegenProperty.isArray) { // array
+            return "[" + constructExampleCode(codegenProperty.items, modelMaps, processedModelMap) + "]";
+        } else if (codegenProperty.isMap) {
+            return "{ key: " + constructExampleCode(codegenProperty.items, modelMaps, processedModelMap) + "}";
+        } else if (codegenProperty.isPrimitiveType) { // primitive type
+            if (codegenProperty.isString) {
+                if (!StringUtils.isEmpty(codegenProperty.example) && !"null".equals(codegenProperty.example)) {
+                    return "\"" + codegenProperty.example + "\"";
```
Comment (jirikuncar) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RubyClientCodegen.java:664]:
```suggestion
                    return "'" + codegenProperty.example + "'";
```

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RubyClientCodegen.java [hunk 2]
```diff
@@ -667,4 +569,178 @@ protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel, Sc
...
+                } else {
+                    return "\"" + codegenProperty.name + "_example\"";
```
Comment (jirikuncar) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RubyClientCodegen.java:666]:
```suggestion
                    return "'" + codegenProperty.name + "_example'";
```

**Comment by zippolyte** [comment]
I think this is ready for review, can you have a look @autopp @wing328 ? Thanks !

**Comment by wing328** [comment]
Pushed https://github.com/OpenAPITools/openapi-generator/pull/8040/commits/7e9f3554b64d4227591c660ca0bb04441f43306b to update the samples so as to fix CircleCI build failure. Let's see how that goes.

**Comment by zippolyte** [comment]
It seemed to do it, thanks 🙂

File: samples/client/petstore/ruby-faraday/docs/FakeApi.md [hunk 3]
```diff
@@ -457,19 +457,19 @@ Petstore.configure do |config|
...
   int32: 56, # Integer | None
-  int64: 56, # Integer | None
+  int64: 789, # Integer | None
   float: 3.4, # Float | None
   string: 'string_example', # String | None
-  binary: File.new('/path/to/file'), # File | None
+  binary: File.new('/path/to/some/file'), # File | None
   date: Date.parse('2013-10-20'), # Date | None
-  date_time: Time.parse('2013-10-20T19:20:30+01:00'), # Time | None
+  date_time: DateTime.parse('2013-10-20T19:20:30+01:00'), # Time | None
```
Comment (wing328) [inline samples/client/petstore/ruby-faraday/docs/FakeApi.md:472]: @zippolyte Shall we use `Time` instead?

Ref: https://github.com/OpenAPITools/openapi-generator/pull/7656

**Comment by wing328** [comment]
cc @cliffano (2017/07) @zlx (2017/09) @autopp (2019/02)

**Comment by wing328** [comment]
@zippolyte when you've time, I wonder if you can resolve the merge conflicts. I'll put other Ruby PRs on hold for the time being. Have a nice weekend.

**Comment by zippolyte** [comment]
@wing328 I rebased, but now I'm not too sure why circle ci is failing.

**Comment by zippolyte** [comment]
Hey @wing328, can you have another look at this one ? Thanks 🙇

**Comment by wing328** [comment]
@zippolyte PR merged. Thanks for your contribution.

**Pull request merged**